src/module.c

```diff
@@ -752,8 +788,11 @@ static inline int add(RedisModuleCtx *ctx,
         return RTS_ReplyGeneralError(ctx, "TSDB: the key is not a TSDB key");
     } else {
         series = RedisModule_ModuleTypeGetValue(key);
+        if (ParseDuplicatePolicy(ctx, argv, argc, &dp) != TSDB_OK) {
+            return REDISMODULE_ERR;
+        }
```

Comment:
Should this be called even if a new series was created?

Reply:
@ashtul its already being parsed in `parseCreateArgs`.